src/domain/safe/safe.repository.ts

```diff
@@ -78,6 +79,30 @@ export class SafeRepository implements ISafeRepository {
     return page;
   }
 
+  async getModuleTransactions(
+    chainId: string,
+    safeAddress: string,
+    to?: string,
+    module?: string,
+    limit?: number,
+    offset?: number,
+  ): Promise<Page<ModuleTransaction>> {
+    const transactionService =
+      await this.transactionApiManager.getTransactionApi(chainId);
+    const page = await transactionService.getModuleTransactions(
+      safeAddress,
+      to,
+      module,
+      limit,
+      offset,
+    );
+    page.results.map((moduleTransaction) =>
+      this.transactionTypeValidator.validate(moduleTransaction),
```

Comment:
Shouldn't the schema fail for other transactions which are not module transactions?

Right now it's using a schema that accepts Ethereum and Multisig transactions. We have a `moduleTransactionTypeSchema` in place already so my suggestion is to have a validator for it and use it here.

Reply:
Yes, I agree, I included a specific validator for `ModuleTransaction` [here]([2bdb7e3](https://github.com/5afe/safe-client-gateway-nest/pull/187/commits/2bdb7e39b2f7da588718aa545f4fe8d2c48ddf1c)) 🙂